pallets/parachain-system/src/lib.rs

```diff
@@ -583,6 +582,12 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(super) type LastDmqMqcHead<T: Config> = StorageValue<_, MessageQueueChain, ValueQuery>;
 
+	/// The index of the next message to be processed from the queue.
+	///
+	/// The value is loaded before and saved after processing messages.
+	#[pallet::storage]
+	pub(super) type NextDmqMessageIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
```

Comment:
using a type wrapper for MessageIndex might help

Reply:
Yeah, that makes a lot of sense to use the one from the polkadot dmp pallet.